l1-contracts/src/core/RollupCore.sol

```diff
@@ -35,29 +35,171 @@ import {StakingQueueConfig} from "@aztec/core/libraries/compressed-data/StakingQ
+ *      - Attestations in blocks are not validated on-chain to save gas. Since attestations are still posted to L1,
+ *        nodes are expected to verify them off-chain, and skip a block if its attestations are invalid.
+ *      - If a block has invalid attestation signatures, anyone can call `invalidateBadAttestation()`
+ *      - If a block has insufficient valid attestations (< 2/3 of committee), anyone can call
+ *        `invalidateInsufficientAttestations()`
+ *      - While anyone can call invalidation functions, it is expected that the next proposer will do so, and if they
+ *        fail to do so, then other committee members do, and if they fail to do so, then any validator will do so.
+ *      - Upon invalidation, the invalid block and all subsequent blocks are removed from the chain, so the pending
+ *        chain tip reverts to the block immediately before the invalid one.
+ *       - Note that only unproven blocks can be invalidated, as proven blocks are final and cannot be reverted.
+ *
+ *      Unhappy path for missing proofs:
+ *      - Each epoch has a proof submission window (configured via aztecProofSubmissionEpochs).
+ *      - If no proof is submitted within the window, it is assumed that the epoch cannot be proven due to missing data,
+ *        so all blocks in the epoch are pruned. This is done by calling `prune()` manually, or automatically on the
+ *        next proposal.
+ *      - The committee for the epoch is expected to disseminate transaction data to allow proving, so a prune is
+ *        considered a slashable offense,
+ *        that causes validators to vote for slashing the committee of the unproven epoch.
+ *      - When the pending chain is pruned, all unproven blocks are removed from the pending chain, and the chain
+ *        resumes from the last proven block.
+ *
+ * @dev Slashing Mechanism
+ *
+ *      Slashing is a critical security mechanism that penalizes validators who misbehave or fail to fulfill their
+ *      duties. The slashing process is governance-based and operates through a voting mechanism:
+ *
+ *      - When nodes detect validator misbehavior, they create a proposal for slashing the offending validators
+ *      - The proposal is submitted to the slashing contract and enters a voting period
+ *      - Each block proposer votes on the slashing proposal during their assigned slot
+ *      - If the proposal receives sufficient votes (reaches the configured quorum), it passes
+ *      - Once approved, the offending validators are slashed, meaning their staked assets are reduced by the slashing
+ *        amount
+ *      - If a validator's stake falls below the minimum required amount due to slashing, they are automatically
+ *        removed from the validator set
+ *
+ *      Conditions that cause nodes to vote for slashing a validator:
+ *      1. Validators that fail to fulfill their duties:
+ *         - Block proposers who fail to propose during their assigned slot
```

Comment:
I find this interesting as this differs from Ethereum, where block proposer who misses a slot just missed the block reward. Why are we aiming for a better liveness than Ethereum? Is it because we can afford it as Aztec's validators will be way fewer and more professional?